### PR TITLE
Datastore operations (adding and searching all)

### PR DESCRIPTION
**Pull request by Mistermakos**

### CODE REVIEW

File: checkers-backend/src/api/game/game.controller.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        try
+        {
+            const query = this.datastore.createQuery('Game')
+            const result = await this.datastore.runQuery(query)
+            return await this.apiService.ApiSuccessData({result})
+        }
+        catch(err){return await this.apiService.ApiErrorData(err);}
+    }
+    
+    // @Get(":id")
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:22]: commented code is a code smell, remove it

File: checkers-backend/src/api/api.service.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+
+
+@Injectable()
+export class ApiService {
+    async ApiSuccessData(data : object): Promise<object>
```
Comment (yahor-mikheyenka-wttech) [inline checkers-backend/src/api/api.service.ts:6]: please, use code formatting for all `.ts` files

File: checkers-backend/src/api/game/game.controller.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    // async FindOne(@Param() id: number ): Promise<object> {
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:23]: remove it, commented code is a code smell

File: checkers-backend/src/api/game/game.controller.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    //     try{
+    //         const taskKey = await this.datastore.key([ 'Game', id ]);
+    //         const result = await this.datastore.createQuery('Game').hasAncestor(taskKey);
+    //         return await this.apiService.ApiSuccessData({result})
+    //     }
+    //     catch(err){return await this.apiService.ApiErrorData(err);}
+    // }
+
+    @Post()
+    async AddOne(@Body() body: object): Promise<string> {
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:33]: just `add` is enough

File: checkers-backend/src/api/game/game.controller.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+import { Datastore } from '@google-cloud/datastore';
+import { take } from 'rxjs';
+
+@Controller('api/v1/Game')
+export class GameController {
+    constructor(private readonly apiService: ApiService) {}
+    datastore = new Datastore({databaseId:'checkers-datastore', projectId:"checkers-zsl"});
+
+    @Get()
+    async FindAll(): Promise<object> {
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:12]: in typescript we should using camelcase, so are methods should beggin with lower-case
https://pl.wikipedia.org/wiki/CamelCase

File: checkers-backend/src/api/game/game.controller.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+              key: taskKey,
+              data: {body},
+            };
+            await this.datastore.save(entity);
+            return await this.apiService.ApiSuccessNoData();
+        }
+        catch(err){return await this.apiService.ApiErrorNoData();}
+    }
+
+//     @Put(':id')
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:47]: remove this code

File: checkers-backend/src/api/game/game.controller.ts (hunk 4 above)
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.ts:33]: it's good idea to create dto classes -> https://medium.com/@yelinliu/dto-explained-in-nestjs-3a296498d77b

File: checkers-backend/src/api/game/game.controller.spec.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/game/game.controller.spec.ts:1]: nice idea to add unit test's to controllers, but currently we are not testing here anything. In future will be nice to test in those tests some logic in module. But you do not need to do  this now :)

**Review by yahor-mikheyenka-wttech — DISMISSED** [review]
ts formatting issue

File: checkers-backend/src/api/game/game.controller.ts (hunk 4 above)
Comment (yahor-mikheyenka-wttech) [reply checkers-backend/src/api/game/game.controller.ts:33]: or just interfaces or even types
https://www.typescriptlang.org/docs/handbook/interfaces.html
https://www.typescriptlang.org/docs/handbook/advanced-types.html

File: checkers-backend/src/api/gamemoves/gamemoves.controller.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import { ApiService } from '../api.service';
+import { Datastore } from '@google-cloud/datastore';
+
+@Controller('api/v1/GameMoves')
+export class GameMovesController {
+    constructor(private readonly apiService: ApiService) {}
+    datastore = new Datastore({databaseId:'checkers-datastore', projectId:"checkers-zsl"});
+
+    @Get()
+    async FindAll(): Promise<object> {
```
Comment (yahor-mikheyenka-wttech) [inline checkers-backend/src/api/gamemoves/gamemoves.controller.ts:11]: as @jan-kowalczyk-wttech wrote above, please use `camel case`

File: checkers-backend/src/api/room/room.controller.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        try {
+            const query = this.datastore.createQuery('Room')
+            const result = await this.datastore.runQuery(query)
+            return await this.apiService.ApiSuccessData({ result })
+        }
+        catch (err) { throw new NotFoundException("Couldn't find all rooms")}
+    }
+
+    @Post()
+    async Add(@Body() body: createRoom): Promise<string> {
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/room/room.controller.ts:22]: add

File: checkers-backend/src/api/gamemoves/gamemoves.controller.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        try {
+            const query = this.datastore.createQuery('GameMoves')
+            const result = await this.datastore.runQuery(query)
+            return await this.apiService.ApiSuccessData({ result })
+        }
+        catch (err) { throw new NotFoundException("Couldn't find all gamemoves")}
+    }
+
+    @Post()
+    async Add(@Body() body: createGameMove): Promise<string> {
```
Comment (jan-kowalczyk-wttech) [inline checkers-backend/src/api/gamemoves/gamemoves.controller.ts:22]: add

**Review by Hayk-Tovmasyan — APPROVED** [review]

**Pull request merged**